src/tk/wurst_client/mods/BaseFinderMod.java

```diff
@@ -60,94 +69,51 @@ public void onUpdate()
 		if(hasTimePassedM(3000))
 		{
 			matchingBlocks.clear();
+			
+			int playerX = (int) Math.floor(mc.thePlayer.posX);
+			int playerY = (int) Math.floor(mc.thePlayer.posY);
+			int playerZ = (int) Math.floor(mc.thePlayer.posZ);
+
 			for(int y = range; y >= -range; y--)
-			{
 				for(int x = range; x >= -range; x--)
-				{
 					for(int z = range; z >= -range; z--)
 					{
-						int posX = (int)(mc.thePlayer.posX + x);
-						int posY = (int)(mc.thePlayer.posY + y);
-						int posZ = (int)(mc.thePlayer.posZ + z);
+						int posX = playerX + x;
+						int posY = playerY + y;
+						int posZ = playerZ + z;
 						BlockPos pos = new BlockPos(posX, posY, posZ);
 						if(!naturalBlocks.contains(mc.theWorld.getBlockState(
 							pos).getBlock()))
 							matchingBlocks.add(pos);
-						if(matchingBlocks.size() >= maxBlocks)
+						if(foundTooManyBlocks())
 							break;
 					}
-					if(matchingBlocks.size() >= maxBlocks)
-						break;
-				}
-				if(matchingBlocks.size() >= maxBlocks)
-					break;
-			}
-			if(matchingBlocks.size() >= maxBlocks && shouldInform)
+			
+			if(foundTooManyBlocks())
 			{
-				wurst.chat.warning(getName() + " found �lA LOT�r of blocks.");
-				wurst.chat
+				if(shouldInform) {
+					wurst.chat.warning(getName() + " found �lA LOT�r of blocks.");
+					wurst.chat
 					.message("To prevent lag, it will only show the first "
 						+ maxBlocks + " blocks.");
-				shouldInform = false;
-			}else if(matchingBlocks.size() < maxBlocks)
+					shouldInform = false;
+				}
+			} else
 				shouldInform = true;
+			
 			updateLastMS();
 		}
 	}
 	
+	private boolean foundTooManyBlocks() {
```

Comment:
Most JIT comilers will start optimizing at ~10 000 calls.
This will be reached before even one tick was executed.

So I would expect that it will be inlined at runtime.